template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/CmTemplateProcessor.java

```diff
@@ -243,7 +250,48 @@ private Map<String, Set<ServiceComponent>> getNonGatewayServicesByHostGroup() {
                                 .filter(i -> !i.getComponent().equalsIgnoreCase("GATEWAY"))
                                 .collect(toSet())
                 ));
+    }
+
+    private Set<String> collectComponentByHostGroupWithRoleDependencies() {
+        Map<String, Set<ServiceComponent>> hgToNonGwServiceComponents = getNonGatewayServicesByHostGroup();
+        Set<String> services = hgToNonGwServiceComponents.values().stream()
+                .filter(serviceComponents -> !Strings.isNullOrEmpty(getDependentService(serviceComponents)))
+                .map(serviceComponents -> ROLE_DEPENDENCY.get(getDependentService(serviceComponents)))
+                .collect(toSet());
+        return services;
+    }
+
+    private String getDependentService(Set<ServiceComponent> serviceComponents) {
+        Set<String> serviceComponentRoles = collectComponents(serviceComponents);
+        return ROLE_DEPENDENCY.keySet().stream().filter(serviceComponentRoles::contains).findFirst().orElse(null);
+    }
+
+    public List<String> getUnhealthyDependentHostGroups(StackDto stackdto) {
+        if (stackdto.getStackVersion().equals("7.2.11") || stackdto.getStackVersion().equals("7.2.10") || stackdto.getStackVersion().equals("7.2.9")
+                || stackdto.getStackVersion().equals("7.2.8") || stackdto.getStackVersion().equals("7.2.7") || stackdto.getStackVersion().equals("7.2.6")
+                || stackdto.getStackVersion().equals("7.2.2") || stackdto.getStackVersion().equals("7.2.1") || stackdto.getStackVersion().equals("7.2.0")) {
+            List<String> noCheckRequired = new ArrayList<>();
+            noCheckRequired.add("No check required");
```

Comment:
Please remove this. `CMRepositoryVersionUtil` has utility methods to do runtime comparisons

Reply:
Okay, I will use method from CMRepositoryVersionUtil.